### PR TITLE
Clean-up colliding 'SubmapData' names.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

File: cartographer/mapping/internal/2d/pose_graph_2d.h [hunk 1]
```diff
@@ -246,14 +246,14 @@ class PoseGraph2D : public PoseGraph {
...
-  MapById<SubmapId, SubmapData> submap_data_ GUARDED_BY(mutex_);
+  MapById<SubmapId, InternalSubmapData> submap_data_ GUARDED_BY(mutex_);
 
   // Data that are currently being shown.
   MapById<NodeId, TrajectoryNode> trajectory_nodes_ GUARDED_BY(mutex_);
   int num_trajectory_nodes_ GUARDED_BY(mutex_) = 0;
 
   // Global submap poses currently used for displaying data.
-  MapById<SubmapId, pose_graph::SubmapData2D> global_submap_poses_
+  MapById<SubmapId, pose_graph::SubmapOptInput2D> global_submap_poses_
```
Comment (gaschler) [inline cartographer/mapping/internal/2d/pose_graph_2d.h]: Well, it also acts as output from the optimization, but fine.

File: cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d.cc [hunk 2]
```diff
@@ -52,8 +52,8 @@ namespace {
 using LandmarkNode = ::cartographer::mapping::PoseGraphInterface::LandmarkNode;
 using TrajectoryData =
     ::cartographer::mapping::PoseGraphInterface::TrajectoryData;
-using NodeData = NodeData3D;
-using SubmapData = SubmapData3D;
+using NodeOptInput = NodeOptInput3D;
```
Comment (gaschler) [inline cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d.cc]: consistency: this you didn't do in optimization_problem_2d.cc

**Comment by pifon2a** [comment]
Rename stuff a bit again.

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 4.

**Pull request merged**